src/axom/inlet/Inlet.cpp

```diff
@@ -179,5 +179,15 @@ bool Inlet::get(const std::string& name, std::string& value)
   return true;
 }
 
+void Inlet::registerDocWriter(std::shared_ptr<DocWriter> writer) {
+  m_docWriter = writer;
+}
+
+void Inlet::writeDocs() {
+  if (m_docWriterToggle) {
```

Comment:
Can you change this to a more descriptive name? What is it toggling? m_writeDocs.. m_isDocWriteMode.. also check that m_docWriter is not nullptr before using it.